### PR TITLE
wtclient: handle un-acked updates for exhausted sessions

### PR DESCRIPTION
**Pull request by ellemouton**
This PR addresses an edge case that could lead to the "tower has un-acked updates" error 
when a user tries to remove a tower. See the issue [here](https://github.com/lightningnetwork/lnd/issues/6397). 

## The Issue:

Currently, if you call "remove tower", then only sessions from that tower that are loaded into memory on 
startup are terminated correctly. And we only load in sessions where `session.SeqNum < session.policy.MaxUpdates`. 
So the issue occurs when the session is technically exhausted _but_ there is still an un-acked update on disk. 
This means that we would not terminate such a session properly on tower removal and so would run into "tower has un-acked updates". 

## The Solution: 

So with this PR, we make sure to load a session into memory if there are still un-acked updates that it needs to deal with. 

## Flow of the PR: 

1. First, the bug is demonstrated through a test.
2. Then, we extend the session fetching filter function to also count the number of un-acked updates so that we can decide to load the session if there are un-acked updates. 
3. The test is then adjusted to show that the bug has been fixed

### CODE REVIEW

**Review by bhandras — APPROVED** [review]
LGTM, super clean and readable change! Very very nice! ✨

File: watchtower/wtclient/client_test.go [hunk 1]
```diff
@@ -2418,13 +2418,13 @@ var clientTests = []clientTest{
...
 	},
 	{
-		// Demonstrate that a client is un-able to remove a tower if
-		// there are persisted un-acked updates in a session that is
-		// not loaded as a candidate session on start-up.
-		//
-		// NOTE: This is a bug and will be fixed in a followup commit.
-		name: "can't remove tower with un-acked update in " +
-			"non-candidate session",
+		// Previously we would not load a session into memory if it's
```
[the remark this answers is not in the document]
Comment (ellemouton) [reply watchtower/wtclient/client_test.go]: hehe silly me 🙈  - fixed!

File: watchtower/wtdb/client_db.go [hunk 2]
```diff
@@ -2584,35 +2613,42 @@ func (c *ClientDB) filterClientSessionAcks(sessionBkt,
...
-// PerCommittedUpdateCB callback.
+// PerCommittedUpdateCB callback. If the countUpdates argument is true, then
+// the number of committed updates will be returned.
 func filterClientSessionCommits(sessionBkt kvdb.RBucket, s *ClientSession,
-	cb PerCommittedUpdateCB) error {
+	cb PerCommittedUpdateCB, countUpdates bool) (uint16, error) {
 
-	if cb == nil {
-		return nil
+	if cb == nil && !countUpdates {
```
Comment (bitromortac) [inline watchtower/wtdb/client_db.go:2621]: If that check was removed, we wouldn't need the boolean option and make the config options easier to deal with, right? Does it save a lot to exit early here?

**Review by bitromortac — COMMENTED** [review]
Nice find and fix :rocket:. I only have a question regarding the `CountCommittedUpdates bool`

File: watchtower/wtdb/client_db.go (hunk 2 above)
Comment (ellemouton) [reply watchtower/wtdb/client_db.go]: you're totally right! Good catch :) removed it 🎉

**Comment by ellemouton** [comment]
@bhandras  - gonna re-request since i've updated as per @bitromortac's comment about not needing the boolean option 🙏

**Review by bhandras — APPROVED** [review]
PR after minor cleanup still LGTM 🌊

File: watchtower/wtclient/client_test.go [hunk 3]
```diff
@@ -2498,14 +2497,22 @@ var clientTests = []clientTest{
 			require.NoError(h.t, h.clientMgr.Stop())
```
[the remark this answers is not in the document]
Comment (ellemouton) [reply watchtower/wtclient/client_test.go:2497]: good catch! thanks :)

**Pull request merged**